### PR TITLE
generic conditional compilation independent of .NET 6.0 target framework

### PR DESCRIPTION
**Pull request by ewerspej**
I have reworked the .NET 6.0 target framework fix to be future proof and work with .NET 7.0 as well. This way, only the actual TargetFrameworks need to be updated and not all the other places where the targets are used for conditional compilation.

### CODE REVIEW

File: src/Plugin.Maui.Audio/Plugin.Maui.Audio.csproj [hunk 1]
```diff
@@ -36,31 +36,37 @@
...
-	<ItemGroup Condition="$(TargetFramework.Contains('-windows')) != true ">
+	<!-- Windows -->
+	<ItemGroup Condition="$(TargetFramework.StartsWith('net')) == true AND $(TargetFramework.Contains('-windows')) != true">
 		<Compile Remove="**\*.windows.cs" />
 		<None Include="**\*.windows.cs" Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder)" />
 		<Compile Remove="**\windows\**\*.cs" />
 		<None Include="**\windows\**\*.cs" Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder)" />
 	</ItemGroup>
+	<!-- .NET (generic) -->
+	<ItemGroup Condition="!($(TargetFramework.StartsWith('net')) == true AND $(TargetFramework.EndsWith('.0')) == true AND $(TargetFramework.Contains('-')) != true)">
```
Comment (bijington) [inline src/Plugin.Maui.Audio/Plugin.Maui.Audio.csproj:65]: is `.0` safe here? I wonder if we could run into issues if `net6.1` or `net7.1`, etc. ships

**Review by bijington — COMMENTED** [review]
@ewerspej thanks for this! Anything that removes a `#if` is fantastic in my book :)

I had one query about the use of checking for a specific number in the target framework, otherwise it looks good

File: src/Plugin.Maui.Audio/Plugin.Maui.Audio.csproj (hunk 1 above)
Comment (ewerspej) [reply src/Plugin.Maui.Audio/Plugin.Maui.Audio.csproj:65]: That's how it's currently done in the MAUI and MAUI Community Toolkit repositories. Of course, skipping the `.0` would be possible, but then we need to check if there is any kind of number and nothing else that follows it.

Comment (jfversluis) [reply src/Plugin.Maui.Audio/Plugin.Maui.Audio.csproj:65]: Yeah let's see about that when we get there :D

**Review by jfversluis — APPROVED** [review]

**Pull request merged**